### DISCUSSION TITLE
Is Python UDTF deployment supported by snowcli?

### DISCUSSION THREAD

**Discussion by py-prash** [Snowflake-Labs/snowcli#182] [Q&A]
Hello!

 I have a bunch of python UDTFs which I would like to deploy to snowflake using snowcli. But I dont see an option to deploy a python udtf using snowcli. Is there a way to deploy udtf?

Regards,
Prashanth

**Comment by py-prash** [comment]
I was able to figure this one out! The only issue is that the pipeline is failing at the describe function(post create).

**Comment by sfc-gh-kgaputis** [comment]
Hi @py-prash ,
You can definitely use snowcli to deploy Python UDTFs!   Just make sure that your `app.toml` file is configured correctly.   The main things are that your return type needs to have the format `table ( columns to return... )` and  the handler needs to be a Python class which defines the `process()` method.

Here is an example: 
```
[default]
input_parameters = "(file_url varchar)"
return_type = "table (file_url varchar, parsed_data variant)"
file = "app.zip"
name = "my_unstructured_parser"
handler = "app.MyUnstructuredParser"
execute_as_caller = true
```

**Reply by py-prash** [reply]
@sfc-gh-kgaputis Thanks for the reply! I was able to get the function created but the pipeline is failing at the describe function post create. I did comment about the issue here https://github.com/Snowflake-Labs/snowcli/issues/72#issuecomment-1590246645.

**Reply by jhollowell** [reply]
Is using the `app.toml` or now the `config.toml` still a valid way of defining the input_parameters, return_type, file, handler, etc..? How do I get the `snow snowpark function create ...` command to recognize that file?
